tests/components/wilight/test_init.py

```diff
@@ -0,0 +1,66 @@
+"""Tests for the WiLight integration."""
+from asynctest import patch
+import pytest
+import pywilight
+
+from homeassistant.components.wilight.const import DOMAIN
+from homeassistant.config_entries import (
+    ENTRY_STATE_LOADED,
+    ENTRY_STATE_NOT_LOADED,
+    ENTRY_STATE_SETUP_RETRY,
+)
+from homeassistant.helpers.typing import HomeAssistantType
+
+from tests.components.wilight import (
+    HOST,
+    UPNP_MAC_ADDRESS,
+    UPNP_MODEL_NAME,
+    UPNP_MODEL_NUMBER,
+    UPNP_SERIAL,
+    setup_integration,
+)
+
+
+@pytest.fixture(name="dummy_create_api_device")
+def mock_dummy_create_api_device():
+    """Mock a valid api_devce."""
+
+    device = pywilight.discovery.wilight_from_discovery(
```

Comment:
Won't this do I/O?

Reply:
My API has a simulation mode for testing that is activated with:

    device.set_dummy (True)

Making the patch:
     with patch (
         "homeassistant.components.wilight.parent_device.create_api_device",
         return_value = device,
     ):
         yield device

In this way I simulate the IOs ...

Reply:
Ok. We should patch our own code, ie `create_api_device`. Patch the library instead to return the dummy device.

Reply:
I'll check how I do this.